src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/SourceTextExtensions.cs

```diff
@@ -15,9 +17,81 @@ public static RazorSourceDocument GetRazorSourceDocument(this SourceText sourceT
                 throw new ArgumentNullException(nameof(sourceText));
             }
 
-            var content = sourceText.ToString();
```

Comment:
Both Roslyn and us were holding onto the string representation of the document. Instead we now wrap the document. Wrote a boat load of tests to ensure both abstractions are equivalent